modules/apps/dataset/dataset-custom-view-service/service.xml

```diff
@@ -0,0 +1,43 @@
+<?xml version="1.0"?>
+<!DOCTYPE service-builder PUBLIC "-//Liferay//DTD Service Builder 7.4.0//EN" "http://www.liferay.com/dtd/liferay-service-builder_7_4_0.dtd">
+
+<service-builder auto-import-default-references="false" auto-namespace-tables="false" dependency-injector="ds" mvcc-enabled="true" package-path="com.liferay.dataset.custom.view">
+	<namespace>DatasetCustomView</namespace>
+	<entity local-service="true" name="DatasetCustomViewEntry" remote-service="false" trash-enabled="false" uuid="true">
+
+		<!-- PK fields -->
+
+		<column container-model="true" name="datasetCustomViewEntryId" primary="true" type="long" />
+
+		<!-- Audit fields -->
+
+		<column name="companyId" type="long" />
+		<column name="userId" type="long" />
+		<column name="userName" type="String" uad-anonymize-field-name="fullName" />
+		<column name="createDate" type="Date" />
+		<column name="modifiedDate" type="Date" />
+
+		<!-- Other fields -->
+
+		<column name="datasetDisplayId" type="String" />
```

Comment:
Why would a `view` have a reference to a `datasetDisplayId`? Shouldn't this be the other way around? (Maybe I'm not reading it correctly)

Reply:
Same for `plid` and `portletId`... that seems to imply `views` won't be able to be reused which I think was something we already corrected in our latest conversations with @dsanz

Reply:
Yes, we're evaluating the best way to tie views to the _data source_ used by the dataset display, rather than to the dataset display instance itself.

Reply:
A summary of what @dsanz and I have been brainstorming is that this API will address CustomViewSettings by (userid, datasetViewId, name) tuples, where name is the name the user gives to the CustomViewSettings when saving it. 

That way, for any CustomViewSettings object we are capable of knowing to which pages of the portal (invocations of the taglib) it may be applied (because of the datasetViewId) and we can have more than one stored configuration for the same datasetViewId (because we have the name field).

Then, for each page in the portal (invocation of the taglib) we will have an "internal" CustomViewSettings which represents the current state of that page which can be the same of a named CustomViewSettings, or an altered version of one of the named CustomViewSettings.

Also, this internal CustomViewSettings can be 1-1 mapped to the current setttings we store in portalPreferences, because we store exactly one per (portlet, layout) tuple, which is exactly the same set we have defined for "internal" CustomViewSettings.

Then, managing CustomViewSettings from the perspective of the user is just a matter of associating CustomViewSettings instances (addressed by datasetViewId and name) to the pages he visits. For that, we will provide another API that defines the active CustomViewSettings for any (userid, datasetViewId, portletId, plid) tuple.